cranelift-wasm/tests/wasm_testsuite.rs

```diff
@@ -45,6 +45,35 @@ fn use_fallthrough_return() {
     );
 }
 
+#[test]
+fn use_name_section() {
+    let wat = read_file(Path::new("../wasmtests/names.wat")).expect("error reading wat file");
+    let data = Wat2Wasm::new()
+        .write_debug_names(true)
+        .convert(wat)
+        .unwrap_or_else(|e| panic!("error converting wat to wasm: {:?}", e));
+
+    let flags = Flags::new(settings::builder());
+    let triple = triple!("riscv64");
+    let isa = isa::lookup(triple).unwrap().finish(flags.clone());
+    let return_mode = ReturnMode::NormalReturns;
+    let mut dummy_environ = DummyEnvironment::new(isa.frontend_config(), return_mode, false);
+
+    translate_module(data.as_ref(), &mut dummy_environ).unwrap();
+
+    assert_eq!(dummy_environ.get_module_name().unwrap(), "module_name");
+    assert_eq!(
+        dummy_environ.get_func_name(FuncIndex::from_u32(0)).unwrap(),
+        "func_name"
+    );
+    assert_eq!(
+        dummy_environ
+            .get_local_name(FuncIndex::from_u32(0), LocalIndex::from_u32(0))
+            .unwrap(),
+        "loc_name"
+    );
```

Comment:
:point_up: wouldn't mind dividing these 3 assertions out into separate tests if needed, but that felt a little noisy for this file, whose main focus is the `testsuite` function.